phoenix-core/src/it/java/org/apache/phoenix/end2end/TableDDLPermissionsIT.java

```diff
@@ -211,23 +211,14 @@ public Void run() throws Exception {
 
             // we should be able to read the data from another index as well to which we have not given any access to
             // this user
-            verifyAllowed(createIndex(indexName2, phoenixTableName), unprivilegedUser);
```

Comment:
@kadirozde - I'm curious what the purpose of these changes are. Have DDL permissions changed? If not, and your changes somehow break the existing test, is there a way to modify the test logic to keep the same test coverage?

Reply:
Currently global indexes are built by using UPSERT SELECT, i..e, by reading mutations from the data table and writing the corresponding mutations to the index table. The user does not need to have the write privilege on the data table in order to build an index table. However, in the new approach, an index table is built by replaying the existing mutations back on the data table again (without actually applying them to the data table) for the purpose of reconstructing the index mutations. Such replay attempts are allowed only when the user has the write privilege. Therefore, I had to delete the scenarios that are not applicable anymore.